Artifacts/windows-ServiceFabricSDK/InstallServiceFabricSDKAndTools.ps1

```diff
@@ -1,136 +1,375 @@
+    )
+
+    # Get correct Service Fabric SDK Product ID for version of VS that is installed.
+    switch ($VSVersionNumber){
+        14 { return 'MicrosoftAzure-ServiceFabric-VS2015' }  #tools and sdk
+        default { return 'MicrosoftAzure-ServiceFabric-CoreSDK' } #core sdk only
     }
 }
 
-### For VS2017, must also enable component for Service Fabric tools
-if ($VSVersion -ge 15){
-    #update the installer
+function Install-ServiceFabricSdk
+{
+    [CmdletBinding()]
+    param(
+        [int] $VSVersionNumber,
+        [string] $LogPath = $(Join-Path $env:Temp 'ServiceFabricSDK.log')
+    )
+
+    $wpiExe = Get-WebPlatformInstaller
+    $productId = Get-ServiceFabricSdkProductId -VSVersionNumber $VSVersionNumber
+    
+    Invoke-Process -FileName $wpiExe -Arguments "/Offline /Products:$productId /Path:$($env:Temp)\OfflineCache"
+    Invoke-Process -FileName $wpiExe -Arguments "/Install /Products:$productId /AcceptEula /SuppressReboot /SuppressPostFinish /Log:$LogPath /xml:$($env:Temp)\OfflineCache\feeds\latest\webproductlist.xml"
+}
+
+function Enable-ServiceFabricTools
+{
+    [CmdletBinding()]
+    param(
+    )
+
     $vsBootstrapperExe = Join-Path $env:Temp "vsbootstrap.exe"
     $vsBootstrapperUrl = 'https://aka.ms/vs/15/release/vs_enterprise.exe'
-    Write-Verbose "Downloading Visual Studio bootstrapper from $vsBootstrapperUrl."
+
+    Write-Host "Downloading Visual Studio bootstrapper from $vsBootstrapperUrl"
     try 
     {
-        (New-Object System.Net.WebClient).DownloadFile($vsBootstrapperUrl, $vsBootstrapperExe )
-    } catch [System.Management.Automation.MethodInvocationException] {
-        Write-Error "Unable to find Visual Studio bootstapper at $vsBootstrapperUrl."
+        (New-Object System.Net.WebClient).DownloadFile($vsBootstrapperUrl, $vsBootstrapperExe)
+    }
+    catch [System.Management.Automation.MethodInvocationException]
+    {
+        throw "Unable to find Visual Studio bootstapper at $vsBootstrapperUrl"
     }
-    if ((Test-Path $vsBootstrapperExe) -eq $false){
-        Write-Error "Visual Studio bootstrapper ($vsBootstrapperUrl) not successfully download to $vsBootstrapperExe."
-        return -1
+
+    if (-not (Test-Path $vsBootstrapperExe))
+    {
+        throw "Visual Studio bootstrapper was not successfully downloaded to $vsBootstrapperExe"
     }
-    StartAndWaitForProcess $vsBootstrapperExe "--quiet --update" @(0)
 
-    #Get path to vs_installer.exe
-    $vsInstallerExePath = (Get-ItemProperty -Path 'Registry::HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Uninstall\{6F320B93-EE3C-4826-85E0-ADF79F8D4C61}' -Name 'InstallLocation').InstallLocation.Trim('"')
+    Write-Host 'Getting Visual Studio Installer path'
+    $vsInstallerExePath = (Get-ItemProperty -Path "Registry::HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Uninstall\{6F320B93-EE3C-4826-85E0-ADF79F8D4C61}" -Name 'InstallLocation').InstallLocation.Trim('"')
```

Comment:
As a general best practice I stick to single quotes for all literal strings. I use double quotes for strings that I want expanded (`"$myVar"`), or where I want to use escape characters (``"Break this over `n two lines"``).

In the case of the `Get-ItemProperty` call, we need to use double quotes so that the `"Registry::HKLM"` gets expanded correctly. Otherwise, the call will fail.